src/zcash/Proof.hpp

```diff
@@ -235,6 +235,42 @@ class ZCProof {
     }
 };
 
+void initialize_curve_params();
+
+class ProofVerifier {
+private:
+    bool perform_verification;
+
+    ProofVerifier(bool perform_verification) : perform_verification(perform_verification) { }
+
+public:
+    // ProofVerifier should never be copied
+    ProofVerifier(const ProofVerifier&) = delete;
+    ProofVerifier& operator=(const ProofVerifier&) = delete;
+    ProofVerifier(ProofVerifier&&);
+    ProofVerifier& operator=(ProofVerifier&&);
+
+    // Creates a verification context that strictly verifies
+    // all proofs using libsnark's API.
+    static ProofVerifier Strict();
+
+    // Creates a dummy verification context that performs
+    // no verification, used when avoiding duplicate effort
+    // such as during reindexing.
+    static ProofVerifier Dummy();
```

Comment:
Since this is actually used in production and not just in tests, can we rename Dummy to NoVerify or NoVerification ?  Dummy doesn't give the code reviewer any information; for all they know, the Dummy ProofViewer might be a mocked out object or a verifier using a dummy curve.